homeassistant/helpers/entity.py

```diff
@@ -1156,6 +1166,9 @@ async def _async_registry_updated(
         await self.async_remove(force_remove=True)
 
         self.entity_id = registry_entry.entity_id
+
+        # Clear the remove event to handle entity added again after entity id change
+        self.__remove_event = None
```

Comment:
This needs to be moved to `add_to_platform_start` unless PR https://github.com/home-assistant/core/pull/102911 is merged